gateway/join_test.go

```diff
@@ -378,10 +278,13 @@ func TestHandleJoin(t *testing.T) {
 	devices[testNodeName] = testDevice
 
 	g := &gateway{
-		logger:   logging.NewTestLogger(t),
-		devices:  devices,
-		dataFile: createDataFile(t),
+		logger:  logging.NewTestLogger(t),
+		devices: devices,
 	}
+	dataDirectory1 := t.TempDir()
+	t.Setenv("VIAM_MODULE_DATA", dataDirectory1)
```

Comment:
Rather than doing this `t.Setenv` here, you could have the `g.setupSqlite` take a second parameter of `pathPrefix string` and not need to stub out the environment variable.